ElectronClient/app/gui/SideBar.jsx

```diff
@@ -440,10 +446,40 @@ class SideBarComponent extends React.Component {
 	makeHeader(key, label, iconName, extraProps = {}) {
 		const style = this.style().header;
 		const icon = <i style={{ fontSize: style.fontSize * 1.2, marginRight: 5 }} className={"fa " + iconName} />;
+
+		if (extraProps.toggleblock || extraProps.onClick) {
+			style.cursor = "pointer";
+		}
+
+		let headerClick = extraProps.onClick || null;
+		delete extraProps.onClick;
+
+		// check if toggling option is set.
+		let toggleIcon = null;
+		const toggleKey = `${key}IsExpanded`;
+		if (extraProps.toggleblock) {
+			let isExpanded = this.state[toggleKey];
+			toggleIcon = <i className={`fa ${isExpanded ? 'fa-chevron-down' : 'fa-chevron-left'}`} style={{ fontSize: style.fontSize * 0.75,
+				marginRight: 12, marginLeft: 5, marginTop: style.fontSize * 0.125}}></i>;
+		}
 		return (
-			<div style={style} key={key} {...extraProps}>
+			<div style={style} key={key} {...extraProps} onClick={async (event) => {
+				// if a custom click event is attached, trigger that.
+				if (headerClick) {
+					await headerClick(event, key);
+				}
+
+				if (extraProps.toggleblock) {
+					const isExpanded = this.state[toggleKey];
+					this.setState({
+						[toggleKey]:	!isExpanded
+					});
+					Setting.setValue(toggleKey, !isExpanded);
```

Comment:
I wanted to isolate the changes to this component itself, hence assigning the `Setting` value here itself. Let me know if that is not appropriate.

Reply:
Yes I think this is fine.